modules/services/way-displays.nix

```diff
@@ -77,35 +95,49 @@ in
   config = mkIf cfg.enable {
     assertions = [
       (lib.hm.assertions.assertPlatform "services.way-displays" pkgs lib.platforms.linux)
+      {
+        assertion = cfg.statefulConfiguration -> cfg.settings == null;
+        message = "way-displays: when `statefulConfiguration = true` then `settings` must be `null`";
+      }
+      {
+        assertion = (!cfg.statefulConfiguration) -> cfg.settings != null;
+        message = "way-displays: `settings` must be non-null unless `statefulConfiguration = true`";
+      }
     ];
 
-    xdg.configFile."way-displays/cfg.yaml".source =
-      yaml.generate "way-displays-config.yaml"
-        (mergeSets [
-          {
-            CALLBACK_CMD = "${pkgs.libnotify}/bin/notify-send \"way-displays \${CALLBACK_LEVEL}\" \"\${CALLBACK_MSG}\"";
-          }
-          cfg.settings
-        ]);
+    home.packages = [ cfg.package ];
```

Comment:
The package being available in the environment is aligned with the stateful workflow but I think it can be useful for the static case as well so I did not make it conditional.